elastic/apm-server
# modelindexer: Report all Stack Monitoring metrics

## Motivation/summary

This patch adds the missing stack monitoring metrics to the APM Server's
new modelindexer.

## Checklist



~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Build the binary locally for linux: `GOOS=linux make; cp apm-server testing/docker/fleet-server/`
2. Check out https://github.com/elastic/apm-server/pull/7453 (Or after the PR has been merged, skip this step).
3. Run `./testing/stack-monitoring.sh`
4. Navigate to `http://localhost:5601` and go to Stack Monitoring.
5. Index a bunch of events:
```
$ while true; do curl -H "Content-type: application/x-ndjson" --data-binary @testdata/intake-v2/spans.ndjson http://localhost:8200/intake/v2/events && sleep 0.1 ; done
```
7. Verify that
  6.1. Output Enabled is `Elasticsearch`
  6.2. Total Events Rate is > 0.0.
  6.3. Bytes Sent Rate > 0.0.

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/7286993/156549448-d12bab77-40e1-441a-bb40-1b760c42dfcf.png">

8. Verify that most libbeat metrics aren't 0:

```console
$ curl -s localhost:8200/debug/vars | grep libbeat.
"libbeat.pipeline.events.total": 9849,
"libbeat.output.write.bytes": 1780,
"libbeat.output.type": "elasticsearch",
"libbeat.output.events.acked": 9849,
"libbeat.output.events.active": 0,
"libbeat.output.events.batches": 178,
"libbeat.output.events.failed": 0,
"libbeat.output.events.toomany": 0,
"libbeat.output.events.total": 9849,
```

## Related issues

Closes #7426 
Closes #7186
